NitroxClient/MonoBehaviours/Gui/Chat/PlayerChat.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using NitroxClient.GameLogic.ChatUI;
+using NitroxModel.Core;
+using UnityEngine;
+using UnityEngine.EventSystems;
+using UnityEngine.UI;
+
+namespace NitroxClient.MonoBehaviours.Gui.Chat
+{
+    public class PlayerChat : uGUI_InputGroup
+    {
+        private const int LINE_CHAR_LIMIT = 255;
+        private const int MESSAGE_LIMIT = 64;
+        private const float TOGGLED_TRANSPARENCY = 0.4f;
+        public const float CHAT_VISIBILITY_TIME_LENGTH = 10f;
+
+        private PlayerChatManager playerChatManager;
+        private CanvasGroup canvasGroup;
+        private HorizontalOrVerticalLayoutGroup[] layoutGroups;
+        private GameObject logEntryPrefab;
+        private Image[] backgroundImages;
+        private bool transparent;
+        private InputField inputField;
+
+        private Dictionary<ChatLogEntry, GameObject> entries = new Dictionary<ChatLogEntry, GameObject>();
```

Comment:
Looks like you don't need a `Dictionary` here. I think a `Queue<GameObject>` would be best or add a `GameObject` field to `ChatLogEntry` and pass the gameobject to it so you can access it later if needed.